File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a,b) => b - a)
    
    let count = 0
    
    let left = 0
    let right = people.length - 1
     
    while(left <= right){        
        if(left < right && people[left] + people[right] <= limit){
            right--
        }
        
        left++
        count++
    }
    
    return count;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int n = people.length;
        
        Arrays.sort(people);
        
        int left = 0;
        int right = n-1;
        int boat = 0;
        while(left <= right){
            int sum = people[right];
            right--;
            if(sum + people[left] <= limit){
                left++;
            }
            boat++;
        }
        
        return boat;
    }
}